### PR TITLE
Move clib-demo to samples

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

Detailed tests included in `clib_test.c` are superseded by the gtest `clib` suite. Nevertheless, the original file contains a useful example. This PR moves a slightly updated example to `samples/clib` while retaining the output file comparison. 

**If applicable, provide an example illustrating new features this pull request is introducing**

The behavior of

```
scons test-clib-demo
```

is unchanged, but the source code is now compiled from `samples/clib/demo.c`.

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

**Comment by ischoegl** [comment]
The updated unit test works, but I'm not sure why the substituted `*.in` files aren't copied over to `build/samples/clib` (working on macOS/Apple Silicon).

File: samples/clib/Makefile.in [hunk 1]
```diff
@@ -0,0 +1,19 @@
+include @make_Cantera_dot_mak@
+
+CC=@CC@
+RM=rm -f
+CXXFLAGS=@mak_compiler_flags@
```
Comment (speth) [inline samples/clib/Makefile.in]: `CXXFLAGS` shouldn't be used here -- should this be `CFLAGS` instead?

**Comment by speth** [comment]
> The updated unit test works, but I'm not sure why the substituted `*.in` files aren't copied over to `build/samples/clib` (working on macOS/Apple Silicon).

I tested this, and they do get generated when you run the `install` command. I think it's because the intermediate files aren't added to the `build` target, but the final installed location is added to the `install` target. The same is true for the sample build scripts for other interfaces, so I don't think there's anything that needs to be changed here.

File: samples/clib/SConscript [hunk 2]
```diff
@@ -0,0 +1,69 @@
+from os.path import join as pjoin, relpath
+from buildutils import *
+
+Import('env', 'install', 'buildSample')
+localenv = env.Clone()
+
+# localenv['mak_stdlib'] = ['-l' + lib for lib in env['cxx_stdlib']]
```
Comment (speth) [inline samples/clib/SConscript]: Commented-out bit can be deleted.

File: SConstruct [hunk 3]
```diff
@@ -2300,10 +2300,13 @@ if env["python_sdist"]:
...
 sampledir_excludes = ['\\.o$', '^~$', '\\.in', 'SConscript']
 SConscript('build/samples/cxx/SConscript')
+SConscript('build/samples/clib/SConscript')
 
-# Install C++ samples
+# Install C++ / C samples
 install(env.RecursiveInstall, '$inst_sampledir/cxx',
         'samples/cxx', exclude=sampledir_excludes)
+install(env.RecursiveInstall, '$inst_sampledir/clib',
+        'samples/clib', sampledir_excludes)
```
Comment (speth) [inline SConstruct]: I think this should be:
```suggestion
install(env.RecursiveInstall, '$inst_sampledir/clib',
        'samples/clib', exclude=sampledir_excludes)
```

**Review by speth — CHANGES_REQUESTED** [review]
Thanks, @ischoegl. I think this change generally makes sense, and installing a pure `clib` example is useful.

One thing that's lost here is what was described at the top of `clib_test.c`:
```c
// Include all clib headers to make sure all of them are C-compatible, even if
// we don't actually use all of them in this test.
#include "cantera/clib/ct.h"
#include "cantera/clib/ctfunc.h"
#include "cantera/clib/ctmultiphase.h"
#include "cantera/clib/ctonedim.h"
#include "cantera/clib/ctreactor.h"
#include "cantera/clib/ctrpath.h"
#include "cantera/clib/ctsurf.h"
```
The point was that this is the only file within Cantera that actually gets compiled by the C (rather than C++) compiler, and ensures that these files don't do anything bad like include a C++ header or syntax that isn't valid C (which was in fact the case before I originally added this test). This is why this test was added to the older `test_problems` test suite rather than the all-C++ gtest test suite, and I think we still need _something_ that ensures this, even if it's not the previous version of the `clib` test.

**Comment by ischoegl** [comment]
@speth ... thanks for the feedback and for shedding some light on the `*.in` files (the `cxx` substitutions make some noise during the `scons build` stage, and I made the wrong conclusions).

> One thing that's lost here is what was described at the top of `clib_test.c`:
> 
> ```c
> // Include all clib headers to make sure all of them are C-compatible, even if
> // we don't actually use all of them in this test.
> #include "cantera/clib/ct.h"
> #include "cantera/clib/ctfunc.h"
> #include "cantera/clib/ctmultiphase.h"
> #include "cantera/clib/ctonedim.h"
> #include "cantera/clib/ctreactor.h"
> #include "cantera/clib/ctrpath.h"
> #include "cantera/clib/ctsurf.h"
> ```

Good point. I re-added the 'lost' includes to `demo.c`, and added a comment explaining why they are there.

**Review by speth — APPROVED** [review]
Looks good to me!

**Pull request merged**